Amplify/Categories/Auth/AuthCategoryClientBehavior.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public protocol AuthCategoryClientBehavior {
+
+    func signUp(username: String,
+                password: String,
+                options: AuthSignUpOperation.Request.Options?,
+                listener: AuthSignUpOperation.EventListener?) -> AuthSignUpOperation
+
+    func confirmSignUp(username: String,
+                       code: String,
+                       options: AuthConfirmSignUpOperation.Request.Options?,
+                       listener: AuthConfirmSignUpOperation.EventListener?) -> AuthConfirmSignUpOperation
+
+    func signIn(username: String,
+                password: String,
+                options: AuthSignInOperation.Request.Options?,
+                listener: AuthSignInOperation.EventListener?) -> AuthSignInOperation
+
+    func signInWithSocial(provider: AuthSocialProvider,
+                          token: String,
+                          options: AuthSocialSignInOperation.Request.Options?,
+                          listener: AuthSocialSignInOperation.EventListener?) -> AuthSocialSignInOperation
+
+    func signInWithUI(options: AuthUISignInOperation.Request.Options?,
+                      listener: AuthUISignInOperation.EventListener?) -> AuthUISignInOperation
+
+    func fetchAuthState(listener: AuthStateOperation.EventListener?) -> AuthStateOperation
+
+    // MARK: - Password Management
+
+    func forgotPassword(username: String,
+                        options: AuthForgotPasswordOperation.Request.Options?,
+                        listener: AuthForgotPasswordOperation.EventListener?) -> AuthForgotPasswordOperation
+
+    func confirmForgotPassword(username: String,
+                               newPassword: String,
+                               confirmationCode: String,
+                               options: AuthConfirmForgotPwdOperation.Request.Options?,
+                               listener: AuthConfirmForgotPwdOperation.EventListener?) -> AuthConfirmForgotPwdOperation
```

Comment:
I think AuthConfirmForgotPasswordOperation is more clear than AuthConfirmForgotPwdOperation.  Having a long name doesn't bother me here since I dont think customers will actually have to write this class name out, right?  Anyways, I don't think this should block you from pushing this into master

Reply:
Changed name to AuthConfirmForgotPasswordOperation